### ISSUE TITLE
Mention `getCurrentPosition` can be listened to only once on iOS + add `hasActivePositionStream`

### ISSUE THREAD

**Issue by Arley011** [Baseflow/flutter-geolocator#1254]
## 🏗 Enhancement Proposal

As mentioned in changelog of version 7.6.0, `getCurrentPosition` method logic was updated. [Here is a PR](https://github.com/Baseflow/flutter-geolocator/pull/826) of the update where mentioned that package now uses "specialized `requestLocation` method".  
Inside its documentation mentioned that method will return nothing if location stream is already listened: 
> Calling this method while location services are already running does nothing.

**So I suggest adding this info to the package's documentation of `getCurrentPosition` method.**

P.S. I also suggest adding a getter for `GeolocatorApple` to let developers know when position stream is listened so that other approach should be used to get position instead of `getCurrentPosition`:
```dart
bool get hasActivePositionStream => _positionStream != null;
```
 
### Pitch

I faced the issue that `getCurrentPosition` never returns anything if timeout is not specified on iOS. So I found why it was not working - because of active stream. 
Adding this info for package's method documentation will save time when developers face the same issue

### Platforms affected (mark all that apply)
- [x] :iphone: iOS
- [ ] :robot: Android

**Comment by moshe5745** [comment]
@Arley011 Nice investigation. This is supposed to be a bug, not Enhancement Proposal.

**Comment by itb55** [comment]
I face the similar issue -
second Geolocator.getServiceStatusStream().listen  doesn't work on iOs.
Everything is ok on Android

**Comment by mjolnir91** [comment]
I think these are related https://github.com/Baseflow/flutter-geolocator/issues/1166

**Comment by TimHoogstrate** [comment]
Dear @Arley011,

I cannot find the mentioned: `Calling this method while location services are already running does nothing.`. I would be pleased if you can point me into the right direction so that i can update the documentation properly. We also briefly discussed the suggested feature regarding the `hasActivePositionStream `. We concluded that the developer already keeps track of the position stream itself. In other words, you already know when the stream is active of not. 

@itb55, 

Can you create another ticket, it seems that it is not directly related to this issue. Furthermore, I cannot reproduce your issue, so remember to add example code to that issue.

Kind regards,
